### ISSUE TITLE
xorg fails to start after bootstrap on Debian 9

### ISSUE THREAD

**Issue by DavidCWGA** [cawilliamson/ansible-gpdpocket#107]
Because of #105, I installed plain Debian 9, and then ran bootstrap after.

Before running bootstrap, xorg seems to start fine, albeit rotated sideways.

Bootstrap completes OK but after rebooting, systemd stops at "Started user manager for UID 117". There's no login prompt and no xorg.

I can still SSH in - and if I do, this is what I find in /var/log/messages:

````
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) LoadModule: "glx"
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) Loading /usr/lib/xorg/modules/extensions/libglx.so
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) Module glx: vendor="X.Org Foundation"
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011compiled for 1.19.2, module version = 1.0.0
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011ABI class: X.Org Server Extension, version 10.0
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) LoadModule: "intel"
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) Loading /usr/lib/xorg/modules/drivers/intel_drv.so
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) Module intel: vendor="X.Org Foundation"
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011compiled for 1.19.0, module version = 2.99.917
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011Module class: X.Org Video Driver
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011ABI class: X.Org Video Driver, version 23.0
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) intel: Driver for Intel(R) Integrated Graphics Chipsets:
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011i810, i810-dc100, i810e, i815, i830M, 845G, 854, 852GM/855GM, 865G,
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011915G, E7221 (i915), 915GM, 945G, 945GM, 945GME, Pineview GM,
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011Pineview G, 965G, G35, 965Q, 946GZ, 965GM, 965GME/GLE, G33, Q35, Q33,
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011GM45, 4 Series, G45/G43, Q45/Q43, G41, B43
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) intel: Driver for Intel(R) HD Graphics
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) intel: Driver for Intel(R) Iris(TM) Graphics
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (II) intel: Driver for Intel(R) Iris(TM) Pro Graphics
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE) No devices detected.
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE)
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: Fatal server error:
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE) no screens found(EE)
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE)
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: Please consult the The X.Org Foundation support
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: #011 at http://wiki.x.org
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]:  for help.
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE) Please also check the log file at "/var/log/Xorg.0.log" for additional information.
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE)
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: (EE) Server terminated with error (1). Closing log file.
Aug 23 20:44:11 lionpocket /usr/lib/gdm3/gdm-x-session[11392]: Unable to run X server
Aug 23 20:44:11 lionpocket gdm3: Child process -11392 was already dead.
Aug 23 20:44:11 lionpocket gdm3: Could not start command '/usr/lib/gdm3/gdm-session-worker': Too many open files
Aug 23 20:44:11 lionpocket gdm3: Child process -11392 was already dead.
````

**Comment by DavidCWGA** [comment]
Deleting /etc/X11/xorg.conf.d/20-intel.conf allows xorg to start - but sideways.

**Comment by reldred** [comment]
Similar issues here ~Latest debian unstable and then bootstrapped the system afterwards; had X before bootstrap lost it after bootstrap. XFCE and LightDM here instead of Gnome+GDM which booted me back to CLI instead of having a hung terminal so slightly different, but trying to do a startx manually resulted in almost identical Xorg error (no devices detected, no screens detected).

Likewise, deleting /etc/X11/xorg.conf.d/20-intel.conf got Xorg working again but my screen is now sideways again and I cannot do an xrandr rotate (like it was before the bootstrap).

**Comment by reldred** [comment]
Figured out how to fix it on my end;

tl;dr: get rid of nomodeset on /etc/defaults/grub and run update-grub. You need mode-setting after you've run the ansible playbooks.

Long story; so, I was using nomodeset and fbcon=rotate:1 to get the stock installers working (since for whatever reason I can't bootstrap an actual ISO), and immediately after install was manually editing grub to add those two items each boot, afterwards setting both options in /etc/defaults/grub followed by an update-grub to save time. This got me into Xorg before Bootrapping the system, but post bootstrap with /etc/X11/xorg.conf.d/20-intel.conf still in place you'll get the no-xorg issues. Removed nomodeset from the grub config, rebooted. lightdm was initially sideways and after a few seconds turned back to normal and the logged in session worked fine. Xrandr rotation now works again too.

Maybe the ansible playbooks could do a quick pass through /etc/defaults/grub to get rid of nomodeset?

**Comment by Awwkaw** [comment]
I know this is half-off topic, i've just installed debian on my gpd pocket, but i cannot bootstrap because i have no internet connection, and using the fix from archwiki (putting a new txt file in `/lib/firmware/brcm` and running `sudo modprobe -r brcmfmac` and `sudo modprobe brcmfmac` and then rebooting, does not give me internet access, so i cannot bootstrap?! how did you bootstrap?

**Comment by DavidCWGA** [comment]
Thanks, removing "modeset" does indeed solve the problem. I had started the Debian installer with nomodeset, which had apparently applied that setting to the installed grub as well, I didn't add it myself.

**Comment by cawilliamson** [comment]
@DavidCWGA 

Two points to make here:

1.) This is caused by the `nomodeset` flag as mentioned above and since the original ISO wasn't bootstrapped (entirely understandable in this case) I'm going to close this one off as wontfix but actually I think the original cause (bootstrap-iso not working) is now fixed. :)

2.) Nice beard!
